Common/src/dual_grid_structure.cpp

```diff
@@ -107,6 +108,9 @@ CPoint::CPoint(unsigned short val_nDim, unsigned long val_globalindex, CConfig *
 
   if ( config->GetGrid_Movement() ) {
     GridVel  = new su2double[nDim];
+    GridVel_Old  = new su2double[nDim];
+    GridVel_n  = new su2double[nDim];
+    GridVel_n1  = new su2double[nDim];
```

Comment:
Do we always need this arrays or just in case of the discrete adjoint ?

Reply:
Wrapped the allocation it in a Discrete Adjoint conditional.

Reply:
To give some background why: In the reverse sweep (say, with dual time stepping 2nd order) always the n-2 timestep is loaded from disk into memory. In that process, Coordinates and GridVels are stored. So compared to the corresponding primal timestep the adjoint had data from 2 timesteps before (...thinking in primal time) if these additional containers and corresponding data pushing wouldn't be there.
I thought it is more convenient than recomputing as all the data is already there in the restart files (and it was faster to code). I might also be beneficial for cases where GridVel is not a fixed function like in rigid motion but depends on the problem itself -> then it can be simply read from the restart file.